src/core/evaluator.js

```diff
@@ -1640,19 +1651,34 @@ var PartialEvaluator = (function PartialEvaluatorClosure() {
                 xObjStateManager.transform(matrix);
               }
 
+              // Enqueue textContent chunk before call for sub-form.
+              enqueueChunk();
+              let sinkWrapper = {
+                cachedChunks: [],
+                enqueue(chunk, size) {
+                  sink.enqueue(chunk, size);
+                  this.cachedChunks.push(chunk);
+                },
+
+                get desiredSize() {
+                  return sink.desiredSize;
+                },
+
+                get ready() {
+                  return sink.ready;
+                },
+              };
               next(self.getTextContent({
                 stream: xobj,
                 task,
                 resources: xobj.dict.get('Resources') || resources,
                 stateManager: xObjStateManager,
                 normalizeWhitespace,
                 combineTextItems,
-              }).then(function (formTextContent) {
-                Util.appendToArray(textContent.items, formTextContent.items);
-                Util.extendObj(textContent.styles, formTextContent.styles);
-
+                sinkWrapper,
```

Comment:
> That isn't the case here, so this line will definitely break things; hence you need to change it to sink: sinkWrapper,

Oh, yes. That is why `text` test is failing. Silly mistake :-)
Thanks for pointing out.